driver/health/event_manager_default.go

```diff
@@ -0,0 +1,51 @@
+package health
+
+import (
+	"context"
+	"errors"
+	"reflect"
+)
+
+type DefaultHealthEventManager struct {
+	evtChan                chan interface{}
+	listenerEventTypeCache map[reflect.Type]Readiness
+}
+
+func NewDefaultHealthEventManager() *DefaultHealthEventManager {
+	return &DefaultHealthEventManager{
+		evtChan:                make(chan interface{}),
+		listenerEventTypeCache: make(map[reflect.Type]Readiness),
+	}
+}
+
+func (h *DefaultHealthEventManager) Dispatch(event interface{}) {
+	select {
+	case h.evtChan <- event:
+	default:
```

Comment:
Isn't this an error? Maybe return an error if the channel is blocked. Alternatively maybe spawn a go routine to dispatch the event later.